src/connection.js

```diff
@@ -771,6 +771,63 @@ class Connection {
         this._onIdle();
     }
 
+    /**
+     * Helper function to send stanzas. The main benefit is for sending
+     * stanzas for which you expect a responding stanza with the same id.
+     *
+     * @param {Element} stanza - The stanza to send.
+     * @param {number} [timeout=3000] - The time specified in milliseconds for a
+     *    timeout to occur, the default is 3000.
+     * @return {Promise<Element|null>} A Promise that resolves to:
+     *   - If the request is successful, it resolves to the response stanza.
+     *   - If the server returns an error, it rejects with the error stanza.
+     *   - If the request times out, it rejects with null.
+     */
+    sendAsync(stanza, timeout = 3000) {
```

Comment:
Thanks for your PR.

AFAIK the IQ stanza is the only one for which you can reasonably expect a responding stanza, and Strophe already has a `sendIQ` method. In fact this method looks largely like a copy/paste of that one except that it also returns a Promise.

For what other stanzas/use-cases besides IQ stanzas do you expect to use this method for?

Reply:
I have added a generic compatibility interface designed to support the more elegant and modern approach of asynchronous programming with Promises. While IQ stanzas are the primary use case, I hope this method can provide flexibility for other types of stanzas, especially in scenarios where asynchronous results need to be handled. For instance, there may be cases where you need to retrieve the unique stanza ID of a message, or when we need to send a message and wait for confirmation.

Reply:
With XMPP, if you send a stanza that requires a confirmation, then you send an IQ stanza. For other types of stanzas you cannot expect any kind of confirmation or return stanza.

This change largely just duplicates the already existing `sendIQ` stanza. If we want to add promises, then we need to add them to the existing methods instead of duplicating them as is done here.

Reply:
Thanks for the suggestion! To avoid potential compatibility issues, I've created a new method that supports both callbacks and promises.

If we decide to modify the existing method, I can update it to check for a callback function. If a callback is provided, it will continue to function as before; if not, it will return a Promise.

Reply:
>  I can update it to check for a callback function. If a callback is provided, it will continue to function as before; if not, it will return a Promise.

Yes I think this will be a better approach. :+1: